pkg/alicloud/client/client.go

```diff
@@ -355,3 +358,79 @@ func (c *vpcClient) DescribeNatGateways(req *vpc.DescribeNatGatewaysRequest) (*v
 func (c *vpcClient) DescribeEipAddresses(req *vpc.DescribeEipAddressesRequest) (*vpc.DescribeEipAddressesResponse, error) {
 	return c.client.DescribeEipAddresses(req)
 }
+
+// ramClient defines the struct of VPC client
+type ramClient struct {
+	client *ram.Client
+}
+
+// NewRAMClient creates a new RAM client with given region, AccessKeyID, and AccessKeySecret
+func (f *clientFactory) NewRAMClient(region, accessKeyID, accessKeySecret string) (RAM, error) {
+	client, err := ram.NewClientWithAccessKey(region, accessKeyID, accessKeySecret)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ramClient{
+		client: client,
+	}, nil
+}
+
+// GetServiceLinkedRole returns service linked role from Alicloud SDK calls.
+func (c *ramClient) GetServiceLinkedRole(roleName string) (*ram.Role, error) {
+	request := ram.CreateGetRoleRequest()
+	request.RoleName = roleName
+	request.SetScheme("HTTPS")
+
+	response, err := c.client.GetRole(request)
+	if err != nil {
+		if isNoPermissionError(err) {
+			return nil, fmt.Errorf("no permission to get service linked role, please grant credentials correct privileges. See https://github.com/gardener/gardener-extension-provider-alicloud/blob/master/docs/usage-as-end-user.md#Permissions")
```

Comment:
We don't want to refer the master branch.

```suggestion
			return nil, fmt.Errorf("no permission to get service linked role, please grant credentials correct privileges. See https://github.com/gardener/gardener-extension-provider-alicloud/blob/v1.21.0/docs/usage-as-end-user.md#Permissions")
```